### PR TITLE
[Wallet]: add passphrase recovery

### PR DESCRIPTION
**Pull request by plam-ml**
## Description 

- Add passphrase recovery
- https://mysten.atlassian.net/browse/APPS-1096

hover-add

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/wallet/src/ui/app/components/menu/content/PasswordInputDialog.tsx [hunk 1]
```diff
@@ -63,66 +69,93 @@ export function PasswordInputDialog({
...
-                        <div className="text-center mt-4">
-                            <Text
-                                variant="pBodySmall"
-                                color="steel-dark"
-                                weight="normal"
-                            >
-                                This is the password you currently use to lock
-                                and unlock your Sui wallet.
-                            </Text>
+                            {horizontalFooter ? (
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/menu/content/PasswordInputDialog.tsx:140]: We should just keep the button and keep it consistent

File: apps/wallet/src/ui/app/components/recovery-passphrase/RecoveryPassphrase.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import { MenuLayout } from '_components/menu/content/MenuLayout';
+import { PasswordInputDialog } from '_components/menu/content/PasswordInputDialog';
+import { useNextMenuUrl } from '_components/menu/hooks';
+import { useAppDispatch } from '_hooks';
+import { loadEntropyFromKeyring } from '_redux/slices/account';
+import { entropyToMnemonic, toEntropy } from '_shared/utils/bip39';
+
+export function RecoveryPassphrase() {
+    const [passwordConfirmed, setPasswordConfirmed] = useState(false);
+    const [mnemonic, setMnemonic] = useState('');
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/recovery-passphrase/RecoveryPassphrase.tsx:18]: Can we change this to be an array of strings like we do in Backup page https://github.com/MystenLabs/sui/blob/main/apps/wallet/src/ui/app/pages/initialize/backup/index.tsx#L51? To void having the whole mnemonic in one string? and make `HideShowDisplayBox` render it in spans?

File: apps/wallet/src/ui/app/components/recovery-passphrase/RecoveryPassphrase.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                    continueLabel="Continue"
+                />
+            </div>
+        );
+    }
+
+    return (
+        <MenuLayout title="Your Recovery Passphrase" back={accountsUrl}>
+            <div className="flex flex-col gap-3">
+                <div className="items-center bg-issue-light flex gap-1.5 px-2.5 py-2 rounded-xl">
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/recovery-passphrase/RecoveryPassphrase.tsx:51]: Could we use Alert here?

File: apps/wallet/src/ui/app/components/recovery-passphrase/RecoveryPassphrase.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                    <Info16 className="text-issue-dark w-4 h-4" />
+                    <div>
+                        <Text color="issue-dark" variant="pBodySmall">
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/recovery-passphrase/RecoveryPassphrase.tsx:54]: And just add the text in 2 divs

File: apps/wallet/src/ui/app/components/recovery-passphrase/RecoveryPassphrase.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import { MenuLayout } from '_components/menu/content/MenuLayout';
+import { PasswordInputDialog } from '_components/menu/content/PasswordInputDialog';
+import { useNextMenuUrl } from '_components/menu/hooks';
+import { useAppDispatch } from '_hooks';
+import { loadEntropyFromKeyring } from '_redux/slices/account';
+import { entropyToMnemonic, toEntropy } from '_shared/utils/bip39';
+
+export function RecoveryPassphrase() {
+    const [passwordConfirmed, setPasswordConfirmed] = useState(false);
+    const [mnemonic, setMnemonic] = useState(['']);
```
Comment (Jordan-Mysten) [inline apps/wallet/src/ui/app/components/recovery-passphrase/RecoveryPassphrase.tsx:17]: Can you just default this state to `null` or something?

Comment (plam-ml) [reply apps/wallet/src/ui/app/components/recovery-passphrase/RecoveryPassphrase.tsx]: Updated. I did not want to type the `useState` but that makes sense to set as null

File: apps/wallet/src/ui/app/components/menu/content/AccountActions.tsx [hunk 6]
```diff
@@ -58,6 +59,12 @@ export function AccountActions({ account }: AccountActionsProps) {
     return (
         <div className="flex flex-row items-center flex-1 gap-1">
             {actionContent}
+            <Link
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/menu/content/AccountActions.tsx:62]: Already discussed in slack just noting it here as well. We should only show the `Export Passphrase` button only for derived accounts

**Review by pchrysochoidis — APPROVED** [review]

**Pull request merged**